controllers/cluster_controller.go

```diff
@@ -164,6 +168,32 @@ func (r *ClusterReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cl
 	return res, kerrors.NewAggregate(errs)
 }
 
+func (r *ClusterReconciler) reconcileMetrics(_ context.Context, c *clusterv1.Cluster) {
+
+	if c.Status.ControlPlaneInitialized {
+		metrics.ClusterControlPlaneReady.WithLabelValues(c.Name, c.Namespace).Set(1)
+	} else {
+		metrics.ClusterControlPlaneReady.WithLabelValues(c.Name, c.Namespace).Set(0)
+	}
+	if c.Status.InfrastructureReady {
+		metrics.ClusterInfrastructureReady.WithLabelValues(c.Name, c.Namespace).Set(1)
+	} else {
+		metrics.ClusterInfrastructureReady.WithLabelValues(c.Name, c.Namespace).Set(0)
+	}
+	// TODO: [wfernandes] pass context here
+	_, err := secret.Get(r.Client, c, secret.Kubeconfig)
+	if err != nil {
+		metrics.ClusterKubeconfigReady.WithLabelValues(c.Name, c.Namespace).Set(0)
+	} else {
+		metrics.ClusterKubeconfigReady.WithLabelValues(c.Name, c.Namespace).Set(1)
+	}
+	if c.Status.ErrorReason != nil || (c.Status.ErrorMessage != nil && *c.Status.ErrorMessage != "") {
```

Comment:
It should be sufficient to just check if `c.Status.ErrorReason` and `c.Status.ErrorMessage` are not `nil`